envbuilder.go

```diff
@@ -456,17 +456,18 @@ func Run(ctx context.Context, opts options.Options) error {
 			}
 			kOpts := &config.KanikoOptions{
 				// Boilerplate!
-				CustomPlatform:    platforms.Format(platforms.Normalize(platforms.DefaultSpec())),
-				SnapshotMode:      "redo",
-				RunV2:             true,
-				RunStdout:         stdoutWriter,
-				RunStderr:         stderrWriter,
-				Destinations:      destinations,
-				NoPush:            !opts.PushImage || len(destinations) == 0,
-				CacheRunLayers:    true,
-				CacheCopyLayers:   true,
-				CompressedCaching: true,
-				Compression:       config.ZStd,
+				CustomPlatform:     platforms.Format(platforms.Normalize(platforms.DefaultSpec())),
+				SnapshotMode:       "redo",
+				RunV2:              true,
+				RunStdout:          stdoutWriter,
+				RunStderr:          stderrWriter,
+				Destinations:       destinations,
+				NoPush:             !opts.PushImage || len(destinations) == 0,
+				CacheRunLayers:     true,
+				CacheCopyLayers:    true,
+				ForceBuildMetadata: opts.PushImage, // Force layers with no changes to be cached, required for cache probing.
```

Comment:
Is there any downside to always enabling this option regardless of `PushImage`?

Reply:
I'm not sure. It'll be an additional cache layer to download during builds vs executing the command and ending up with the same result. Mainly added it to be cautious.

Imagine the following scenario:

```dockerfile
RUN if [[ $(date +%s) -gt 1734998400 ]]; then touch "/'member-christmas?"; fi
```

With force build metadata, we would still use the cache here and not produce a new image.

Reply:
Fair enough, let's leave as-is for now 👍